backend.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"errors"
+	"log"
+)
+
+type Backend interface {
+	BackendReader
+	BackendWriter
+	DelayedBackend
+}
```

Comment:
Why use nested interfaces here?

Reply:
When I was first designing this it was necessary to do a little trick. It is no longer relevant and I should remove it. I had separate writers and readers that allowed me to do some things on the backend, some things local. I redesigned it. Good catch.